ddtrace/tracer/option.go

```diff
@@ -511,7 +511,9 @@ func newConfig(opts ...StartOption) *config {
 	}
 	// if using stdout or traces are disabled, agent is disabled
 	agentDisabled := c.logToStdout || !c.enabled.current
+	ignoreStatsdPort := c.agent.ignore // preserve the value of c.agent.ignore when testing
 	c.agent = loadAgentFeatures(agentDisabled, c.agentURL, c.httpClient)
+	c.agent.ignore = ignoreStatsdPort
```

Comment:
The comment tells that we preserve the value of `c.agent.ignore` when testing. I assume that in production services it won't ignore the port reported by the agent because `c.agent.ignore` will be false always when assigned to `ignoreStatsdPort`.

Did I understand it right?

Reply:
yes! the method used to set `c.agent.ignore` is internal, so unless we set it ourselves, it shouldn't be used. I did also leave a warning on the `withIgnoreAgent()` so that future contributors know to only use it for testing.